templates/types/streaming/nextjs/app/components/ui/chat/chat-sources.tsx

```diff
@@ -4,17 +4,60 @@ import { Button } from "../button";
 import { HoverCard, HoverCardContent, HoverCardTrigger } from "../hover-card";
 import { SourceData, SourceNode } from "./index";
 import { useCopyToClipboard } from "./use-copy-to-clipboard";
+import PdfDialog from "./widgets/PdfDialog";
 
 const SCORE_THRESHOLD = 0.5;
 
+function SourceNumberButton({ index }: { index: number }) {
+  return (
+    <div className="text-xs w-5 h-5 rounded-full bg-gray-100 mb-2 flex items-center justify-center hover:text-white hover:bg-primary hover:cursor-pointer">
+      {index + 1}
+    </div>
+  );
+}
+
+enum NODE_TYPE {
+  URL,
+  LOCAL_FILE,
+}
+
+function getNodeInfo(node: SourceNode) {
```

Comment:
In the coude below, you're mapping each time calling `getNodeInfo(node)` - you can just map once, if you include the missing data from `SourceNode` - I think it's just `id` and you can use a type, e.g.:

```suggestion
type NodeInfo {
  type: NodeType;
  path: string;
  id: string;
}

function getNodeInfo(node: SourceNode): NodeInfo {
```